### ISSUE TITLE
TypeScript doesn't report incorrect type when returning AxiosResponse but function declares some different return type

### ISSUE THREAD

**Issue by josephearl** [microsoft/TypeScript#56532]
### 🔎 Search Terms

axios, return, no type check

### 🕗 Version & Regression Information

The app was using TS 4.9.5, but I updated to 5.3.2 and it seems the same.

### ⏯ Playground Link

https://codesandbox.io/p/sandbox/axios-ts-bug-gty2pg?file=%2Findex.ts%3A20%2C1

### 💻 Code

```ts
interface SomeBody {
  field1: string;
  field2: string;
}

async function fetchSomething(): Promise<SomeBody > {
  // Buggy code is missing a call to .data on the AxiosResponse
  return await axios.get('http://example.com/some/api'); // TypeScript is fine, but it should not be
}
```

### 🙁 Actual behavior

It does not fail type checking, however if you introduce an intermediate variable it does:

```ts
async function fetchSomething(): Promise< SomeBody > {
  const response = await axios.get('http://example.com/some/api');
  return response; // TS2322: Type  AxiosResponse<any, any>  is not assignable to type SomeBody
}
```

### 🙂 Expected behavior

It fails type checking because the wrong type is returned by `axios.get` for the return type declared by `fetchSomething`

**Comment by josephearl** [comment]
Thanks I will report it over there
